app/views/spree/admin/shared/_order_summary.html.erb

```diff
@@ -80,7 +80,7 @@
           <strong><%= Spree.t(:payment) %>:</strong>
         </td>
         <td id='payment_status'>
-          <% if @order.payments.count > 0 && @order.payments.last.source.present? && @order.payments.last.source_type == "Spree::MolliePaymentSource" %>
+          <% if @order.payments.count > 0 && @order.payments.last.source_type == "Spree::MolliePaymentSource" && @order.payments.last.source.present?  %>
```

Comment:
After giving this some thought, I think it's better (and more fail-safe) to [use a scope for this](https://guides.rubyonrails.org/active_record_querying.html#scopes). 

Could you add a scope to the `payment_decorator.rb` and factor the views to use this scope? A scope name like `:mollie_source` would make sense. This scope could include all of the checks above (and can be reused multiple times).

Reply:
Yes, sure that's a good idea. 
In general, I thought about refactoring this whole thing to use deface since the overriding the whole order view does not play well with other gems which do that. And also if client apps override this view instead of using deface this view will not be used.  
With that being said, I didn't have the time to look into this since we've been busy lately with going live with a new webshop for a major clothing brand. 

I will look into the scope thing first since we need this update to fix an issue for a client. 

As for the deface, please give me your thoughts on the subject.

Reply:
@vernondegoede  

Once I got into it, I am not so sure about the scope solution. The scope is OK but it might not be enough and optimal.
There are a couple of things that are kinda fishy. 
1. The scopes are usually used to filter multiple records on an association. Adding a scope would require us to chain it to all the order.payments.last( order.payments.mollie_source.last). 
This "train wreck" appears quite often in these views and we would need to litter the whole view with the extra scope. (please see screenshot)
2. And more important, we might be introducing a subtle bug. Using the :mollie_source scope we would get the last Mollie payment for the order. But that is not necessarily the last payment for the order. For websites with different gateways, the backend might display irrelevant and wrong info. 

Maybe a better way to approach this is to add an instance method on the order. Something like: 

`def current_mollie_payment  #or maybe last_mollie_payment 
    payments.mollie_source.last
 end
`
And then use this method in deface overrides. This will clean up the screenshot part a bit.  That way we add Mollie payments display functionality to a site but also let other gateways display properly. 


![image](https://user-images.githubusercontent.com/5826738/49230010-fd2ab280-f3ee-11e8-99c6-1c7b3261cbe5.png)